urllib3/util/retry.py

```diff
@@ -295,6 +294,20 @@ def __repr__(self):
                 'read={self.read}, redirect={self.redirect})').format(
                     cls=type(self), self=self)
 
+    def redirect_method(self, method, status):
+        """
+        Assuming we're doing a redirect, should we change HTTP methods?
+        """
+        if method == 'GET':
+            return method
+        if method == 'HEAD':
+            return method
+        if status == 303:
+            return 'GET'
+        if method == 'POST' and status in set([301, 302, 303]):
```

Comment:
You could in principle remove 303 from this, though it doesn't matter much. It may be better, however, to hoist this set up to module scope (maybe as `_REDIRECT_STATUSES`?) rather than construct it each time we call this function.